src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const obj = Object.assign({}, state);
+  const result = [];
+
+  actions.forEach(element => {
+    switch (element.type) {
+      case 'clear':
+        for (const key in obj) {
+          delete obj[key];
+        }
+        result.push({ ...obj });
```

Comment:
you can push just once in the end of each iteration. There is no sense in duplicating it in every case